src/razor/src/Completion/RazorCompletionItemProvider.ts

```diff
@@ -204,6 +208,13 @@ export class RazorCompletionItemProvider
 
             item = newItem;
 
+            // The documentation object Roslyn returns can have a different
+            // shape than what the client expects, so we do a conversion here.
+            const markdownString = <vscode.MarkdownString>(item.documentation);
```

Comment:
This is weird... and probably against my understanding of typescript. We're casting to `vscode.MarkdownString` to make a new one? Does something magic happen in the constructor?

Reply:
Typescript seems to be weird in that we can have a `vscode.MarkdownString` but the properties on the type may not be what is expected. 

For example, at the beginning of this method we get passed something that _technically_ passes as a `vscode.MarkdownString`, but has totally different properties:
![image](https://github.com/dotnet/vscode-csharp/assets/16968319/d4649185-ae7e-48d8-a2c3-d7c4b613f25e)

After the cast above and the creation of a new `MarkdownString` in the lines below, we're able to return the shape VS Code expects:
![image](https://github.com/dotnet/vscode-csharp/assets/16968319/d9acad37-7226-401d-a606-8e64299f00a4)

Reply:
I suspect this is actually wrong, but not in a way that makes a difference. I would guess that Roslyn LSP is returning the LSP type [MarkupContent](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markupContentInnerDefinition), which has a kind and a value. This code is casting it to a `vscode.MarkdownString` which also has a value, so it can read the value property without a compilation error. Since types, and therefore casts, don't really exist in the compiled JavaScript this works fine. The creation of the new markdown string is then really a conversion from the LSP type to the VS Code type.

Reply:
Appreciate the insight! That would make sense actually. I pushed a change that checks for `MarkupContent` instead, it's a little hacky since we have to convert to `unknown` first, but it's probably better than how it was before.